server.js

```diff
@@ -0,0 +1,218 @@
+          res.end();
+        }
+      });
+      res.end();
+    }
+  });
+});
+
+server.listen(PORT, () => {
+  console.log(`Server started on PORT: ${PORT}`);
+});
+
+/*************** FUNCTIONS **************/
+
+// Add and/or Update element.html
+function addOrUpdateElem(chunk, req, res) {
+  // Used for new element pages
+  let htmlTemplate = `<!DOCTYPE html>
+        <html lang="en">
+          <head>
+            <meta charset="UTF-8" />
+            <title>The Elements - ${chunk["elementName"]}</title>
+            <link rel="stylesheet" href="/css/styles.css" />
+          </head>
+          <body>
+            <h1>${chunk["elementName"]}</h1>
+            <h2>${chunk["elementSymbol"]}</h2>
+            <h3>Atomic number ${chunk["elementAtomicNumber"]}</h3>
+            <p>
+              ${chunk["elementDescription"]}
+            </p>
+            <p><a href="/">back</a></p>
+          </body>
+        </html>`;
+  // Create new element page
+  fs.writeFile(
+    `./public/${chunk["elementName"]}.html`,
+    htmlTemplate,
+    err => {
+      if (err) {
+        return console.log("Cannot write file " + err);
+      }
+    }
+  );
+  res.writeHead(200, {
+    "Content-Type": `text/html, charset=utf-8`,
+    "Content-Length": htmlTemplate.length,
+    "Date": date
+  });
+  res.end();
+}
+
+// Creates and/or updates index.html
+function parseNewHTML(chunk, req, res) {
+  let existingFiles = [];
+  let contentLength = 0;
+  fs.readdir('./public/', (err, files) => {
+    if (err) {
+      return console.log("Directory is not there " + err)
+    }
+    for (let x = 0; x < files.length; x++) {
+      if (!['.keep', '404.html', 'css', 'index.html'].includes(files[x])) {
+        existingFiles.push(files[x]);
+      }
+    };
+    // Top part of index.html
+    let indexTop = `<!DOCTYPE html>
+    <html lang="en">
+    
+    <head>
+      <meta charset="UTF-8" />
+      <title>The Elements</title>
+      <link rel="stylesheet" href="/css/styles.css" />
+    </head>
+    
+    <body>
+      <h1>The Elements</h1>
+      <h2>These are all the known elements.</h2>
+      <h3>These are ${existingFiles.length}</h3>
+      <ol>`;
+    // Middle part of index.html
+    for (let y = 0; y < existingFiles.length; y++) {
+      let indexMid = `
+        <li>
+          <a href="/${existingFiles[y]}">${existingFiles[y].split('.')[0]}</a>
+        </li>`;
+      indexTop += indexMid
+    };
+    // Bottom part of index.html
+    let indexEnd = `
+      </ol>
+    </body>
+  </html>`;
+    contentLength = indexTop.length + indexEnd.length;
+    // index.html is re-written with new ref to new elements
+    fs.writeFile(
+      "./public/index.html",
+      `${indexTop}${indexEnd}`,
+      err => {
+        if (err) {
+          return console.log("Cannot write file " + err);
+        }
+      }
+    );
+  });
+  res.writeHead(200, {
+    "Content-Type": `text/html, charset=utf-8`,
+    "Content-Length": contentLength,
+    "Date": date
+  });
+  res.end();
```

Comment:
res.writeHead/end should be inside the final fs.writeFile so that you respond to the client only after you know you've finished writing the final file.